yal/tests/test_wa_fb_crossover_feedback.py

```diff
@@ -0,0 +1,155 @@
+import json
+from datetime import datetime, timedelta
+
+import pytest
+from sanic import Sanic, response
+
+from vaccine.testing import AppTester, TState, run_sanic
+from yal import config
+from yal.utils import BACK_TO_MAIN, GET_HELP
+from yal.wa_fb_crossover_feedback import Application
+
+
+@pytest.fixture
+def tester():
+    return AppTester(Application)
+
+
+def get_rapidpro_contact(urn):
+    contact = {
+        "uuid": "b733e997-b0b4-4d4d-a3ad-0546e1644aa9",
+        "name": "Test Human",
+        "language": "eng",
+        "groups": [],
+        "fields": {},
+        "blocked": False,
+        "stopped": False,
+        "created_on": "2015-11-11T08:30:24.922024+00:00",
+        "modified_on": "2015-11-11T08:30:25.525936+00:00",
+        "urns": [urn],
+    }
+    if urn == "whatsapp:27820001001":
+        contact["fields"] = {
+            "last_mainmenu_time": f"{datetime.now()-timedelta(days=2)}"
+        }
+    return contact
+
+
+@pytest.fixture
+async def rapidpro_mock():
+    Sanic.test_mode = True
+    app = Sanic("mock_rapidpro")
+    tstate = TState()
+
+    @app.route("/api/v2/contacts.json", methods=["POST"])
+    def update_contact(request):
+        tstate.requests.append(request)
+        return response.json({}, status=200)
+
+    async with run_sanic(app) as server:
+        url = config.RAPIDPRO_URL
+        config.RAPIDPRO_URL = f"http://{server.host}:{server.port}"
+        config.RAPIDPRO_TOKEN = "testtoken"
+        server.tstate = tstate
+        yield server
+        config.RAPIDPRO_URL = url
+
+
+@pytest.mark.asyncio
+async def test_state_not_saw_recent_facebook(tester: AppTester, rapidpro_mock):
+    tester.setup_state("state_wa_fb_crossover_feedback")
+    await tester.user_input("No, I didn't")
+    tester.assert_state(None)
+    assert len(rapidpro_mock.tstate.requests) == 1
+    assert [r.path for r in rapidpro_mock.tstate.requests] == ["/api/v2/contacts.json"]
+
+    post_request = rapidpro_mock.tstate.requests[0]
+
+    engaged_on_facebook = json.loads(post_request.body.decode("utf-8"))["fields"][
+        "engaged_on_facebook"
```

Comment:
pls check both fields that you're setting in this test.
You can use this test for an example of how to do that, complete with a mocked datetime
https://github.com/praekeltfoundation/vaccine-eligibility/blob/main/yal/tests/test_pleasecallme.py#L495

Reply:
The miro board flow doesnt update the recommended content if the user did not see the facebook post, so I didnt update that field in that flow, should I?

Reply:
No that's fine